numpy/lib/type_check.py

```diff
@@ -386,20 +392,21 @@ def nan_to_num(x, copy=True):
     """
     x = _nx.array(x, subok=True, copy=copy)
     xtype = x.dtype.type
+
+    isscalar = (x.ndim == 0)
+
     if not issubclass(xtype, _nx.inexact):
-        return x
+        return x[()] if isscalar else x
```

Comment:
No need to even check `isscalar` - `x[()]` works in all cases

Reply:
Well, removing `if isscalar else x` broke `test_type_check.py:369` :x

Reply:
> by copying instead of replacing in-place

That's not true - it did replace in-place, but it created a new view into the original before returning. I supose that's still a mostly-unjustifiable compatibility change though, so what you have is fine.

I suppose it's possible that something downstream is relying on equality by idenit